### PR TITLE
Added Delaware Transformer

### PR DESCRIPTION
**Pull request by squidarth**
Hey all! I'm Sid (new here). Excited to contribute to this awesome project.

# What's in this PR?

Wrote the Delaware transformer. There are a few fields in the Delaware voter file that I need to do some research on (there were some that were kinda ambiguous and it wasn't clear what the mapping to the fields NVF uses was), and I need to do some more manual testing, but this is more or less ready for review.

Refer [here](http://delawarevoters.info/downloads.html) for Delaware-related voter file information. I used [this csv](http://delawarevoters.info/download/20150521/ActiveReg.csv) as a reference while implementing this.

### Fields I need to do research on 

* I assumed `RD` refers to the Congressional District
* I assumed the `CNLEVY` field (County/ Levy Council District) is the `COUNTY_BOARD_DIST`
* I'm not totally sure what `COUNTY_VOTER_REF` is, I don't there is one
* There's a field called `ED` (election district). I'm not sure what this is referring to, I think it might be relevant to a state house district, but not totally sure
* There's no `PRECINCT` given in the voter file

### Some questions

* For the `PARTY` column, should I go ahead and edit the fixed types to include DE-specific ones?

Hopefully I'm on the right track here, thank you for taking a look!

## TODO

- [x] Finish adding the fake data (in `faker_data.py`)
- [ ] Finish doing research on unknown fields 
- [ ] Manually spot-check a bunch of rows in the DE voter registry after running the transform.

### CODE REVIEW

File: src/python/national_voter_file/tests/test_transformers.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
     BaseTransformer.col_type_dict.keys()
 )
 
-TEST_STATES = ['co', 'fl', 'mi', 'nc', 'ny', 'oh', 'ok', 'wa']
+TEST_STATES = ['co', 'fl', 'mi', 'nc', 'ny', 'oh', 'ok', 'wa', 'de']
```
Comment (slavingia) [inline src/python/national_voter_file/tests/test_transformers.py]: put closer to front, as its supposed to be alphabetical :)

File: src/python/national_voter_file/us_states/de/transformer.py [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+                                                   BaseTransformer)
+import usaddress
+
+__all__ = ['default_file', 'StatePreparer', 'StateTransformer']
+
+default_file = 'DE_Sample.csv'
+
+class StatePreparer(BasePreparer):
+    state_path = 'de' # Two letter code for state
+    state_name='Delaware' # Name of state with no spaces. Use CamelCase
```
Comment (slavingia) [inline src/python/national_voter_file/us_states/de/transformer.py]: spaces around = and other stylistic things were caught by travis. https://travis-ci.org/national-voter-file/national-voter-file/jobs/202202474

**Comment by slavingia** [comment]
yes, i believe you can edit the parties to match DE's file. each state is slightly different.

**Comment by slavingia** [comment]
i think RD is the state house (lower house), SD is the state senate (upper house), and ED is the congressional district

**Comment by squidarth** [comment]
> i think RD is the state house (lower house), SD is the state senate (upper house), and ED is the congressional district

Cool, just based on some of the other states?

Updated rest of PR per your comments.

**Comment by slavingia** [comment]
i was looking at http://delawarevoters.info/download/20140617/Electronic%20Rec%20File%20Layout%20v2.pdf and just prior context on how other state's talk about the house/senate districts.

**Comment by squidarth** [comment]
@slavingia cool, got it, tx

**Comment by BlueMoon3000** [comment]
@slavingia @squidarth I don't think that ED refers to Congressional District. I was trying to figure out how to tackle this for the VT transformer (which I'm working on) -- both Delaware and Vermont have  only a single At-large congressional district. I did some searching and it seems like in DE Election District refers to a sub-district of the Representative District. There's seems to be a list of them here: http://electionsncc.delaware.gov/ed/2012_ed/2012_maps.shtml

So the next question is how should we handle the `CONGRESSIONAL_DIST` field for states with only a single district? My understanding is that it would be incorrect to refer to them as the 1st district since they are distinct from that. I was going to just go with `AL` for the VT transformer, but I don't know if we already have a standard approach for this/if it's OK for this field to be a string.

**Comment by slavingia** [comment]
How are they referred to in the voter file itself? Maybe we should standardize "0" if it's the only district? Let's move this to the Slack.

**Pull request merged**